src/types.ts

```diff
@@ -34,18 +32,31 @@ export interface Config {
   provider: ethers.providers.Provider;
 }
 
+export type Maybe<T> = T | undefined;
+
 export interface Listing {
   price: number;
   holder: string;
 }
 
-export interface ZAuctionInstances {
-  [registrarAddress: string]: zAuction.Instance;
+export interface Map<T> {
```

Comment:
Just wondering, why this over the default typescript map?
e.g. 
`let myMap = new Map<string, CoinGeckoPrice>();
`

Reply:
I hadn't really considered using a full `Map` object, just because all we need is this an interface not any of the more advanced functions that come with a fully-fledged object, but this can do similar effectively as well.